tenseal/tensors/ckksvector.cpp

```diff
@@ -200,4 +202,20 @@ CKKSVector& CKKSVector::mul_plain_inplace(vector<double> to_mul) {
     return *this;
 }
 
+CKKSVector CKKSVector::matmul(const vector<vector<double>>& matrix) {
+    CKKSVector new_vector = *this;
+    return new_vector.matmul_inplace(matrix);
```

Comment:
is there a reason for not using this->matmul_inplace directly?

Reply:
this->matmul_inplace will modify the current object (inplace) while we are try to return a new object here

Reply:
my bad, true, I was thinking about the inplace method.